javascript/code-challenges/challenges-05.test.js

```diff
@@ -0,0 +1,373 @@
+
+{
+  item: 'switch'
+  purchasePrice: 399
+}
+
+------------------------------------------------------------------------------------------------ */
+
+const addPurchases = (arr) => {
+  let totalPurchase = arr.reduce((currentT, currentP) => {
+    return currentT + currentP.purchasePrice;
+  }, 0);
+  return totalPurchase;
+};
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named countNumberOfElements that, given an array as input, uses reduce to count the number of elements in the array.
+
+Note: You may not use the array's built-in length property.
+------------------------------------------------------------------------------------------------ */
+
+const countNumberOfElements = (arr) => {
+  let numberOfElements = arr.reduce((length) => {
+    return ++length; // could also use index and return index + 1;
+  }, 0);
+  return numberOfElements;// Solution code here...
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named returnNames that, given the Star Wars data, below, uses reduce to return an array containing the names of the characters.
+------------------------------------------------------------------------------------------------ */
+
+let starWarsData = [{
+  name: 'Luke Skywalker',
+  height: '172',
+  mass: '77',
+  hair_color: 'blond',
+  skin_color: 'fair',
+  eye_color: 'blue',
+  birth_year: '19BBY',
+  gender: 'male',
+},
+{
+  name: 'C-3PO',
+  height: '167',
+  mass: '75',
+  hair_color: 'n/a',
+  skin_color: 'gold',
+  eye_color: 'yellow',
+  birth_year: '112BBY',
+  gender: 'n/a'},
+{
+  name: 'R2-D2',
+  height: '96',
+  mass: '32',
+  hair_color: 'n/a',
+  skin_color: 'white, blue',
+  eye_color: 'red',
+  birth_year: '33BBY',
+  gender: 'n/a'
+},
+{
+  name: 'Darth Vader',
+  height: '202',
+  mass: '136',
+  hair_color: 'none',
+  skin_color: 'white',
+  eye_color: 'yellow',
+  birth_year: '41.9BBY',
+  gender: 'male'
+},
+{
+  name: 'Leia Organa',
+  height: '150',
+  mass: '49',
+  hair_color: 'brown',
+  skin_color: 'light',
+  eye_color: 'brown',
+  birth_year: '19BBY',
+  gender: 'female'
+}];
+
+const returnNames = (arr) => {
+  let newArr = arr.reduce ((newArr, para) => {
+    newArr.push(para.name);
+    return newArr;
+  }, []);
+  return newArr;// Solution code here...
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+Write a function named reversedString that takes in a string and returns a string with the letters in reverse order.
+
+Note: You must use reduce for this challenge. You may not use the built-in .reverse() string method.
+------------------------------------------------------------------------------------------------ */
+
+const reversedString = (str) => {
+  let bwStr = str.split('').reduce((prevL, curL) => {
```

Comment:
Really nice work on your challenges! 💯 

Feedback here has to do with variable name: `bwStr`. I think I can deduce what this represents (backwardString) based off instructions provided in challenge, but naming it something super straightforward is best so I don't have to take any time guessing what it means :)